### PR TITLE
Up/refactoring

### PR DESCRIPTION
**Pull request by florianajir**

### CODE REVIEW

File: Provider/ApiProvider.php [hunk 1]
```diff
@@ -104,35 +104,44 @@ public function setShowroom(string $showroom): void
...
+        ]);
+        switch ($response->getStatusCode()) {
+            case Response::HTTP_FORBIDDEN:
+                $this->logger->warning('ApiProvider request access denied');
+
+                throw new ApiAccessDeniedException();
+            case Response::HTTP_REQUESTED_RANGE_NOT_SATISFIABLE:
+                $this->logger->warning('ApiProvider request out of range');
+
+                throw new NotFoundHttpException('Out of range');
```
Comment (unikoke94) [inline Provider/ApiProvider.php:129]: https://www.php.net/manual/fr/class.outofrangeexception.php
Maybe more appropriate ?

Comment (florianajir) [reply Provider/ApiProvider.php:129]: I used HttpExceptionInterface to be handled more easily onKernelExeptionListener, if you please to create a listener in frontoffice and showroom so handle other kind of exceptions lets do it

Comment (florianajir) [reply Provider/ApiProvider.php:129]: https://symfony.com/doc/current/event_dispatcher.html#creating-an-event-listener

Comment (florianajir) [reply Provider/ApiProvider.php]: updated

**Pull request merged**